medusa/search/proper.py

```diff
@@ -334,37 +336,52 @@ def _download_propers(self, proper_list):
 
             main_db_con = db.DBConnection()
             history_results = main_db_con.select(
-                b'SELECT resource FROM history '
+                b'SELECT resource, proper_tags FROM history '
                 b'WHERE showid = ? '
                 b'AND season = ? '
-                b'AND episode = ? '
+                b'AND episode IN ({episodes}) '
                 b'AND quality = ? '
                 b'AND date >= ? '
-                b'AND action in (?, ?, ?, ?)',
-                [cur_proper.indexerid, cur_proper.actual_season, cur_proper.actual_episode, cur_proper.quality,
+                b'AND action IN (?, ?, ?, ?) '
+                b'LIMIT {amount}'.format(
+                    episodes=','.join(cur_proper.actual_episodes),
+                    amount=len(cur_proper.actual_episodes) * 2
+                ),
+                [cur_proper.indexerid, cur_proper.actual_season, cur_proper.quality,
                  history_limit.strftime(History.date_format),
                  DOWNLOADED, SNATCHED, SNATCHED_PROPER, SNATCHED_BEST])
 
-            # make sure that none of the existing history downloads are the same proper we're trying to download
-            # if the result exists in history already we need to skip it
-            clean_proper_name = self._canonical_name(cur_proper.name, clear_extension=True)
-            if any(clean_proper_name == self._canonical_name(cur_result[b'resource'], clear_extension=True)
-                   for cur_result in history_results):
-                log.debug('This proper {result!r} is already in history, skipping it', {'result': cur_proper.name})
-                continue
-            else:
-                # make sure that none of the existing history downloads are the same proper we're trying to download
-                clean_proper_name = self._canonical_name(cur_proper.name)
-                if any(clean_proper_name == self._canonical_name(cur_result[b'resource'])
-                       for cur_result in history_results):
-                    log.debug('This proper {result!r} is already in history, skipping it', {'result': cur_proper.name})
-                    continue
+            proper_tags_len = len(cur_proper.proper_tags)
+            proper_name = self._canonical_name(cur_proper.name, clear_extension=True)
+            proper_name_ext = self._canonical_name(cur_proper.name)
+
+            for result in history_results:
+
+                proper_tags = result[b'proper_tags']
+                if proper_tags and len(proper_tags.split('|')) >= proper_tags_len:
```

Comment:
This check makes sure that the current release isn't already proper and in case it is, it will only get the new one if it has a higher proper priority (e.g. PROPER vs REAL.PROPER).